strapi/strapi
# Rename audit logs table name

### What does it do?

- Rename audit log table from `audit_logs` to `strapi_audit_logs`
- Auto migrates the table when Strapi detects the `admin::audit-logs` content type changes the collection name from `audit_logs` to `strapi_audit_logs`

### Why is it needed?

- To be consistent with other internal Strapi tables that use the `strapi_` prefix
- To help users customers who were previously using a custom-made audit logs solution to migrate to the official one

### How to test it?

1st check that audit logs work as usual: make actions in the admin, make sure you see them in audit logs.

Then test the migration:

- to simulate being a 4.6.0 user, change your audit log collection name [here](https://github.com/strapi/strapi/blob/fix/audit-log-table/packages/providers/audit-logs-local/lib/content-types/audit-log/schema.js#L5) to `audit_logs`, as it was before, and restart the server. All previous audit logs will be deleted, but this is expected.
- make an action in the admin, you should see it in audit logs
- then, to simulate the upgrade to 4.6.1, change the collection name again to `strapi_audit_logs` and restart the app
- go to the audit logs, make sure you see your previous action, AND that it has the user attached to it
- make another action in the admin, check that shows up in audit logs as usual
